pcss-criminal-application/src/main/java/ca/bc/gov/open/pcsscriminalapplication/service/impl/CrownValidatorImpl.java

```diff
@@ -43,4 +46,172 @@
+        }
+
+        if(request.getDetail().isEmpty()) {
+            errors.add("Detail is not valid");
+        } else {
+
+            for(int i = 0; i < request.getDetail().size(); i++) {
+
+                // TODO: Add operation mode cd validation
+
+                if(request.getDetail().get(0).getCounselLastNm() == null) {
+                    errors.add(MessageFormat.format("Details CounselLastNm at index {0} is not valid", i+1));
+                }
+
+                if(request.getDetail().get(0).getCounselFirstNm() == null) {
+                    errors.add(MessageFormat.format("Details CounselFirstNm at index {0} is not valid", i+1));
+                }
+
+            }
+
+        }
+
+        return errors;
+
+    }
+
+    @Override
+    public List<String> validateSetCrownAssignment(SetCrownAssignmentRequest request) {
+
+        if (request == null) return Collections.singletonList("Empty request is invalid");
+
+        List<String> errors = new ArrayList<>();
+
+        if(checkAgencyIdentifierTypeForErrors(request.getRequestAgencyIdentifierId())) {
+            errors.add("RequestAgencyIdentifierId is not valid");
+        }
+
+        if(checkSystemIdTypeForErrors(request.getRequestPartId())) {
+            errors.add("RequestPartId is not valid");
+        }
+
+        if(request.getRequestDtm() == null || checkDateTimeTypeForErrors(request.getRequestDtm().toString())) {
+            errors.add("RequestDtm is not valid");
+        }
+
+        if(request.getJustinNo() != null && checkJustinNoTypeForErrors(request.getJustinNo())) {
+            errors.add("JustinNo is not valid");
+        }
+
+        if(request.getCrownAssignment().isEmpty()) {
+            errors.add("CrownAssignment is not valid");
+        } else {
+
+            for (int i = 0; i < request.getCrownAssignment().size(); i++) {
+
+                if (checkAgencyIdentifierTypeForErrors(request.getCrownAssignment().get(i).getAssigningPaasAgencyId())) {
+                    errors.add(MessageFormat.format("CrownAssignment AssigningPaasAgencyId at index {0} is not valid", i+1));
+                }
+
+                if (checkSystemIdTypeForErrors(request.getCrownAssignment().get(i).getAssigningPaasPartId())) {
+                    errors.add(MessageFormat.format("CrownAssignment AssigningPaasPartId at index {0} is not valid", i+1));
+                }
+
+                if (checkSystemSeqNoTypeForErrors(request.getCrownAssignment().get(i).getAssigningPaasSeqNo())) {
+                    errors.add(MessageFormat.format("CrownAssignment AssigningPaasSeqNo at index {0} is not valid", i+1));
+                }
+
+                if (request.getCrownAssignment().get(i).getAssignmentDt() == null ||
+                        checkDateTimeTypeForErrors(request.getCrownAssignment().get(i).getAssignmentDt().toString())) {
+                    errors.add(MessageFormat.format("CrownAssignment AssignmentDt at index {0} is not valid", i+1));
+                }
```

Comment:
You may want to double check the .xsd files for SetCrownAssignment. From what I could tell, null actually appears to be a valid value for all of the CrownAssignment object fields except for OperationModeCd. I haven't confirmed webmethods' behaviour though.

Reply:
Yep that is my plan. I want to go through all the endpoints and confirm validation in one go. There is probably alot of commonality that will be more efficient to work on it that way.